engine/classes/Elgg/ViewsService.php

```diff
@@ -54,11 +54,46 @@ public function __construct(\Elgg\PluginHooksService $hooks, \Elgg\Logger $logge
 		$this->hooks = $hooks;
 		$this->logger = $logger;
 	}
+	
+	/**
+	 * Takes a view name and returns the canonical name for that view.
+	 * 
+	 * @param string $alias The possibly non-canonical view name.
+	 * 
+	 * @return string The canonical view name.
+	 */
+	public function canonicalizeViewName(/*string*/ $alias) /*: string*/ {
+		if (!is_string($alias)) {
+			return false;
+		}
+		
+		$canonical = $alias;
+
+		if (strpos($canonical, "js/") === 0) {
+			$canonical = substr($canonical, 3);
+			if (!pathinfo($canonical, PATHINFO_EXTENSION)) {
+				$canonical .= ".js";
+			}
+		} else if (strpos($canonical, "css/") === 0) {
+			$canonical = substr($canonical, 4);
+			if (!pathinfo($canonical, PATHINFO_EXTENSION)) {
+				$canonical .= ".css";
+			}
+		}
+		
+		if ($alias !== $canonical) {
+			$message = "The deprecated view $alias was moved to $canonical.\n";
```

Comment:
We can retract the warning. I just figured in general aliasing ==
complicated so using canonical name is "better".

On Mon, Jun 22, 2015, 4:13 AM Steve Clay <notifications@github.com> wrote:

> In engine/classes/Elgg/ViewsService.php
> <https://github.com/Elgg/Elgg/pull/8526#discussion_r32923872>:
>
> > +		$canonical = $alias;
> > +
> > +		if (strpos($canonical, "js/") === 0) {
> > +			$canonical = substr($canonical, 3);
> > +			if (!pathinfo($canonical, PATHINFO_EXTENSION)) {
> > +				$canonical .= ".js";
> > +			}
> > +		} else if (strpos($canonical, "css/") === 0) {
> > +			$canonical = substr($canonical, 4);
> > +			if (!pathinfo($canonical, PATHINFO_EXTENSION)) {
> > +				$canonical .= ".css";
> > +			}
> > +		}
> > +		
> > +		if ($alias !== $canonical) {
> > +			$message = "The deprecated view $alias was moved to $canonical.\n";
>
> Do we have a strong motivation for dropping support in 3.0?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/8526/files#r32923872>.
>